scripts/version-harvester

```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+HARVESTER_VERSION=$(cd $1; source ./scripts/version &> /dev/null; echo $VERSION)
```

Comment:
Let's try to avoid polluting the envvars. Does it help if we do `sh` instead of `source` here?

Reply:
Hmmm but because `scripts/version` does not return (echo) anything, I think the parent shell can't get anything from subshell if we replace `source` with `sh`? In other words, the `echo $VERSION` won't work.

Reply:
yep, that doesn't work.
From my testing, envs in the command substitution will not affect the caller context, is it neccessary to `unset DIRTY` in scripts/version?

```
VERSION=v-old
DIRTY=
cat <<EOF > a.sh
DIRTY=1
VERSION=v1
EOF
HARVESTER_VERSION=$(source a.sh &> /dev/null; echo $VERSION)
echo "VERSION: $VERSION"
echo "DIRTY: $DIRTY"
echo "HARVESTER_VERSION: $HARVESTER_VERSION"

# result
VERSION: v-old
DIRTY:
HARVESTER_VERSION: v1
```

Reply:
If `DIRTY` was set in parent shell (which is true in our case because we sourced a version script), subshell will receive the variable:
```bash
VERSION=v-old
DIRTY="itsdirty"  # Set by sourcing a version script in harvester-installer
cat <<EOF > a.sh
VERSION="v1.2.3-${DIRTY}"
EOF
HARVESTER_VERSION=$(source a.sh &> /dev/null; echo $VERSION)
echo "HARVESTER_VERSION: $HARVESTER_VERSION"

# result
HARVESTER_VERSION: v1.2.3-itsdirty
```

In the `version` script, we append `$DIRTY` directly:
```bash
if [[ -z "$DIRTY" && -n "$GIT_TAG" ]]; then
    VERSION=$GIT_TAG
else
    VERSION="${COMMIT}${DIRTY}"
fi
```

Reply:
Oh BTW with or without `DIRTY` in this PR does not affect the behavior. It's only problematic when we source those version scripts in specific order.